src/RsaAlgorithm.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+namespace HttpSignatures;
+
+class RsaAlgorithm implements AlgorithmInterface
+{
+    /** @var string */
+    private $digestName;
+
+    /**
+     * @param string $digestName
+     */
+    public function __construct($digestName)
+    {
+        $this->digestName = $digestName;
+    }
+
+    /**
+     * @return string
+     */
+    public function name()
+    {
+        return sprintf('rsa-%s', $this->digestName);
+    }
+
+    /**
+     * @param string $key
+     * @param string $data
+     *
+     * @return string
+     */
+    public function sign($key, $data)
+    {
+        switch ($this->digestName) {
+          case 'sha256':
+            $algo = OPENSSL_ALGO_SHA256;
+            break;
+          case 'sha1':
```

Comment:
Shouldn't this `sha1` be replaced with the 384 and 512 that are listed in `src/Algorithm.php` above?

Reply:
Indeed, and since 384 and 512 are not listed as options in the spec I've aligned all in the PR branch